### PR TITLE
AT-36 Body html

### PR DESCRIPTION
**Pull request by quarterto**

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -43,8 +45,15 @@ instrument-products:
...
-	./scripts/test.sh
+mocha-opts := --require async-to-gen/register
+
+test: lint $(js-files) $(test-files-all)
+	NODE_ENV=test istanbul cover node_modules/.bin/_mocha -- $(mocha-opts) $(test-files)
+
+unit-test: $(js-files) $(test-files-all)
+	NODE_ENV=test istanbul cover node_modules/.bin/_mocha -- $(mocha-opts) -i --grep "amp validator" $(test-files)
+
+test/amp-transform/%.js: server/stylesheets/main.xsl server/stylesheets/amp/%.xsl
```
Comment (rowanbeentje) [inline makefile]: Could go says you

File: readme.md [hunk 2]
```diff
@@ -25,6 +25,23 @@ Tests
...
 To run the tests and linter locally, run `npm test`.
 
+A Day in the Life of an AMP Article
+---
+
+1. Google crawls FT.com and finds an article, `/content/ffffffff-ffff-ffff-ffff-ffffffffffff`
+2. The article has a `<link rel="amphtml">`, with href pointing to `amp.ft.com/content/ffffffff-ffff-ffff-ffff-ffffffffffff`
+3. The [amp-page.js](server/controllers/amp-page.js) controller runs
+    1. It calls [get-article.js](server/lib/article/get-article.js) to fetch the content from Elastic Search
+    2. It runs various [transforms](server/lib/transforms) on the [article data](server/lib/transforms/article.js) and [body HTML(server/lib/transforms/body.js)
```
Comment (rowanbeentje) [inline readme.md]: borken md on the body HTML link

(2 could probably also use a "...which Google then crawls" suffix)

File: server/lib/live-blogs/transform-live-blogs.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+const cheerioTransform = require('../cheeerio-transform');
```
Comment (rowanbeentje) [inline server/lib/live-blogs/transform-live-blogs.js]: This isn't quite right ;)  The requires on lines 6 and 7 now also need updating - suspect live blogs need testing against Alphaville come Monday

Comment (rowanbeentje) [inline server/lib/live-blogs/transform-live-blogs.js]: (+ some form of even-very-minimal live blogs mock+test?)

File: server/lib/transforms/html/video.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+const url = require('url');
+
+const youtube = videoId => `<amp-youtube
+	data-videoid="${videoId}"
+	layout="responsive"
+	width="480" height="270">
+</amp-youtube>`;
+
+module.exports = match({
+	'a[href^="http://video.ft.com/"]'(el, i, $, {brightcoveAccountId, brightcovePlayerId} = {}) {
```
Comment (rowanbeentje) [inline server/lib/transforms/html/video.js]: This will be a bit invalid for new videos, since they're now ft.com/content/uuid 😢

**Comment by rowanbeentje** [comment]
Thanks for reorganising the folders - much clearer now.

Haven't compared all the transforms with their xlst equivalents, but overall looks good!

File: server/lib/live-blogs/transform-live-blogs.js (hunk 3 above)
Comment (quarterto) [inline server/lib/live-blogs/transform-live-blogs.js]: doh, this file isn't even being used anywhere. which suggests live blogs are fine going through the rest of the pipeline? i'll add tests

File: server/lib/transforms/html/video.js (hunk 4 above)
Comment (quarterto) [inline server/lib/transforms/html/video.js]: → https://jira.ft.com/browse/AT-877

**Pull request merged**